packages/core/src/Engine.ts

```diff
@@ -543,6 +551,35 @@ export class Engine extends EventDispatcher {
           camera._cameraIndex = index;
         }
       );
+
+      const uiCanvases = componentsManager._uiCanvasesArray[CanvasRenderMode.ScreenSpaceOverlay]?._elements;
+      if (uiCanvases) {
+        const {
+          _canvas: canvas,
+          _uiRenderQueue: uiRenderQueue,
+          _virtualCamera: virtualCamera,
+          _renderContext: renderContext,
+          _batcherManager: batcherManager
+        } = this;
+        const { elements: projectE } = virtualCamera.projectionMatrix;
+        const { elements: viewE } = virtualCamera.viewMatrix;
+        (projectE[0] = 2 / canvas.width), (projectE[5] = 2 / canvas.height), (projectE[10] = 0);
+        this._hardwareRenderer.activeRenderTarget(null, new Vector4(0, 0, 1, 1), renderContext.flipProjection, 0);
+        for (let i = uiCanvases.length - 1; i >= 0; i--) {
```

Comment:
UICanvas is sorted in ascending order based on sortOrder，the larger the sortOrder, the later it is rendered. So should handle ui canvas from 0